crawl-ref/source/dat/des/portals/bazaar.des

```diff
@@ -1548,6 +1548,65 @@ xT.xxxxxxxxxxx.Tx
 xxxxxxxxxxxxxxxxx
 ENDMAP
 
+# 3 to 4 shops, with an average of 3.5
+NAME:       bazaar_kenran_forest_clearing
+TAGS:       no_rotate no_vmirror
+ORIENT:     encompass
+WEIGHT:     5
+KITEM:      h = human skeleton
+KFEAT:      $ = any shop
+KFEAT:      # = antique weapon shop / antique armour shop / \
+                antiques shop / book shop
+KFEAT:      A = abandoned_shop
+: if crawl.one_chance_in(4) then
+KFEAT:      F = V
+TILE:       t = dngn_tree_dead w:10 / dngn_tree_red w:3 / \
+                dngn_tree_lightred w:2
+COLOUR:     . = brown, t = brown w:10 / red w:3 / lightred w:2
+: else
+TILE:       t = dngn_tree w:3 / dngn_tree_yellow w:2 / \
+                dngn_tree_lightred w:2 / dngn_tree_red w:3
+KFEAT:      F = T
+COLOUR:     . = green, t = green w:3 / lightgreen w:2 / \
+                lightred w:3 / red w:2 / brown w:1
+: end
+NSUBST:     $ = 1=$A
+SUBST:      $ = #:20 $
+:           bazaar_setup(_G)
+MAP
+ttttttttttttttttttttttttt
+tttttttttttttttttttttt>tt
+ttttttttttttttttttttt.ttt
+ttttttttttttttttttttt.ttt
+ttttttttttttttGttt.t..ttt
+ttttttttttttttt.ttt..tttt
+tttt$.ttttttttt.....ttttt
+tt.t...ttttttt..t.ttttttt
+tt...ttttttt..ttttttt..tt
+tttt....ttt...tt..tt....t
+tttttt.t.t....t..t...t.$t
+ttttt..t..t.F..tttttttttt
+tttttt........t..Attttttt
+tttttttt.tt..t.t..ttttttt
+tttttt....t..t..htt.$tttt
+tttttt.t...t..t.tt...tttt
+ttt......t.....t....ttttt
+ttG.t..tt....tt...ttttttt
+t......tttt..tttttttttttt
+t$..t.tttttt....ttttttttt
+ttt....ttt...ttGttttttttt
+ttttttttt...ttttttttttttt
+ttttttttt..tttttttttttttt
+tttttttttt...tttttttttttt
+ttttttttttt..tttttttttttt
+ttttttttttt.t.ttttttttttt
+tttttttttt..ttttttttttttt
+ttttt.tt...tttttttttttttt
+ttt<.....tttttttttttttttt
+ttttttt.ttttttttttttttttt
+ttttttttttttttttttttttttt
+ENDMAP
```

Comment:
Since you're going for a sorta chaotic random look, you could consider replacing some of the `t`s on the edge of the map with floor tiles at random, so that it doesn't always have the same shape every time. That's not an important feature or requirement, just something that seemed to fit with your current design.